pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -328,51 +333,111 @@ func (a *ApplicationRestoreController) getNamespacedObjectsToDelete(restore *sto
 	return tempObjects, nil
 }
 
-func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.ApplicationRestore) error {
-	restore.Status.Stage = storkapi.ApplicationRestoreStageVolumes
-	if restore.Status.Volumes == nil || len(restore.Status.Volumes) == 0 {
-		backup, err := storkops.Instance().GetApplicationBackup(restore.Spec.BackupName, restore.Namespace)
+func (a *ApplicationRestoreController) updateRestoreCRWithRetry(
+	namespacedName types.NamespacedName,
+	status storkapi.ApplicationRestoreStatusType,
+	stage storkapi.ApplicationRestoreStageType,
+	reason string,
+	volumeInfos []*storkapi.ApplicationRestoreVolumeInfo,
+) (*storkapi.ApplicationRestore, error) {
+	restore := &storkapi.ApplicationRestore{}
+	var err error
+	for i := 0; i < maxRetry; i++ {
+		err := a.client.Get(context.TODO(), namespacedName, restore)
 		if err != nil {
-			return fmt.Errorf("error getting backup spec for restore: %v", err)
+			time.Sleep(retrySleep)
+			continue
 		}
-		backupVolumeInfoMappings := make(map[string][]*storkapi.ApplicationBackupVolumeInfo)
-		objectMap := storkapi.CreateObjectsMap(restore.Spec.IncludeResources)
-		info := storkapi.ObjectInfo{
-			GroupVersionKind: metav1.GroupVersionKind{
-				Group:   "core",
-				Version: "v1",
-				Kind:    "PersistentVolumeClaim",
-			},
+		logrus.Infof("Updating restore  %s/%s in stage/stagus: %s/%s to volume stage", restore.Namespace, restore.Name, restore.Status.Stage, restore.Status.Status)
```

Comment:
Can you handle check for stale update meaning if cr in application stage don't update back it to volume stage

Reply:
Yes will do.

Reply:
you will need to check `ApplicationStage` too